src/OpenCensus.Collector.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -43,7 +46,9 @@ public override void OnStartActivity(Activity activity, object payload)
 
             var request = context.Request;
 
-            this.Tracer.SpanBuilder("HttpIn").SetSampler(this.Sampler).StartScopedSpan();
+            var ctx = this.propagationComponent.TextFormat.Extract<HttpRequest>(request, (r, name) => r.Headers["name"].ToString());
+
+            this.Tracer.SpanBuilderWithRemoteParent("HttpIn", ctx).SetSampler(this.Sampler).StartScopedSpan();
```

Comment:
set the context on the span?

Reply:
`.SpanBuilderWithRemoteParent("HttpIn", ctx)`

Reply:
I have a test for this =)